### ISSUE TITLE
erroneous ReundantCondition + return type issue w/ SplFileObject::fgetcsv()

### ISSUE THREAD

**Issue by WDD-Marv** [vimeo/psalm#665]
The main issue here is that psalm reports while(is_array($csv)) as a redundant condition, on the assumption that `SplFileObject::fgetcsv()` always returns an array.

The second issue is one of the return type; [the docs for SplFileObject::fgetcsv()](https://secure.php.net/manual/en/splfileobject.fgetcsv.php) say "Returns an indexed array containing the fields read, or FALSE on error":

1) psalm reports the return type as `array<mixed, mixed>`, although I'm thinking it should probably be `array<int, string|null>`
2) [3v4l.org indicates the failure return type is NULL, not false](https://3v4l.org/Y7g3v), meaning the overall return type should be `array<int, string|null>|falsey` ?

```php
<?php
$file = new SplTempFileObject(0);

$csv = $file->fgetcsv();

while (is_array($csv)) {
  echo var_export($csv, true);
  
  $csv = $file->fgetcsv();
}

echo var_export($csv, true);
```

**Comment by WDD-Marv** [comment]
now getting `PossiblyNullArgument`, fixiable by doing `if ( ! is_array($thing)) { throw new Exception('foo'); }` :D
